pkg/virt-operator/application.go

```diff
@@ -297,3 +351,58 @@ func (app *VirtOperatorApp) AddFlags() {
 
 	app.AddCommonFlags()
 }
+
+func (app *VirtOperatorApp) getSelfSignedCert() error {
+	var ok bool
+
+	caKeyPair, _ := triple.NewCA("kubevirt.io")
+	keyPair, _ := triple.NewServerKeyPair(
+		caKeyPair,
+		"kubevirt-operator-webhook."+app.operatorNamespace+".pod.cluster.local",
```

Comment:
I see we use this at other places as well, so I would not block on this for this PR, but be aware that `cluster.local` is a cluster setting which can be modified at cluster creation. A starting point would be to at least use a const everywhere (e.g. from `pkg/util/net/dns/resolveconf.go`). And in a next step find out how to get the actual domain from the running cluster.

Reply:
That is a very good point. I wonder if it can also be looked up somewhere from within the cluster. I guess it would be at least possible to read it from the pods resolv.conf.

Thanks for looking.

Reply:
Created a follow up: https://github.com/kubevirt/kubevirt/issues/2998